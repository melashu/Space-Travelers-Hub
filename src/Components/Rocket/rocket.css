.rockets {
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  padding: 1rem;
}

.rocket-list {
  display: flex;
  row-gap: 1.5rem;
}

.banner {
  flex: 1;
  width: 200px;
  height: 200px;
  padding-right: 1rem;
}

.rocket-list .list {
  flex: 4;
  display: flex;
  flex-direction: column;
}

.rocket-button{
    width: max-content;
}

.rocket-name {
    color: #00D7C6;
}

.desc {
  text-align: justify;
}
