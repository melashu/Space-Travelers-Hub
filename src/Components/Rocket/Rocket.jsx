import React, { useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Badge, Button } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import {
  cancleRocket,
  rocketThunk,
  getRocketState,
  reserveRocket,
} from '../../Store/rocketReducer';
import './rocket.css';

const Rocket = () => {
  const rockets = useSelector(getRocketState);
  const dispatch = useDispatch();
  useEffect(() => {
    if (rockets.length === 0) dispatch(rocketThunk());
  }, [dispatch, rockets.length]);
  const rocketsList = rockets.map((rocket) => (
    <div className="rocket-list" key={rocket.id}>
      <img className="banner" src={rocket.flickr_images[0]} alt="banner" />
      <div className="list">
        <h2 className="rocket-name">{rocket.rocket_name}</h2>
        <p className="desc">
          {rocket.reserved ? <Badge>Resereved</Badge> : null}
          {rocket.description}
        </p>
        {rocket.reserved ? (
          <Button
            variant="secondary"
            className="rocket-button"
            size="small"
            onClick={() => {
              dispatch(cancleRocket(rocket.id));
            }}
          >
            Cancel Reservation
          </Button>
        ) : (
          <Button
            className="rocket-button"
            onClick={() => {
              dispatch(reserveRocket(rocket.id));
            }}
            size="small"
          >
            Reserve rocket
          </Button>
        )}
      </div>
    </div>
  ));
  return <div className="rockets">{rocketsList}</div>;
};
export default Rocket;
