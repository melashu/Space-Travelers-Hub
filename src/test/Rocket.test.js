import React from 'react';
import { Provider } from 'react-redux';
import store from '../Store/Store'
import Rocket from '../Components/Rocket/Rocket';
import '@testing-library/jest-dom/extend-expect';

const TestRenderer = require('react-test-renderer'); // ES5 with npm

const RocketsProvider = () => (
  <Provider store={store}>
    <Rocket />
  </Provider>
);

describe('Check Homepage', () => {
  it('renders correctly', () => {
    const component = TestRenderer.create(<RocketsProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
