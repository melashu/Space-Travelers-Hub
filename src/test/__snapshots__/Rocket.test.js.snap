// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Homepage renders correctly 1`] = `
<div
  className="rockets"
/>
`;
